### PR TITLE
HDDS-8201. Added snapshot delete in all tests of TestOmSnapshotFileSystem

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
There are two tasks asked in [HDDS-8201](https://issues.apache.org/jira/browse/HDDS-8201). In #5518, TestOmSnapshotFileSystem tests were migrated to JUnit-5. This change is to address the second task and make some remaining tasks Snapshot delete aware. 

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8201

## How was this patch tested?
Workflow run on fork: https://github.com/hemantk-12/ozone/actions/runs/6872953402/job/18692239021

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java [hunk 1]
```diff
@@ -291,27 +269,46 @@ public void testListKeysAtDifferentLevels() throws Exception {
...
+
+    deleteSnapshot(snapshotName);
+    String expectedMessage = String.format("Unable to load snapshot. " +
+            "Snapshot with table key '/%s/%s/%s' is no longer active",
+        volumeName, bucketName, snapshotName);
+    OMException exception = assertThrows(OMException.class,
+        () -> ozoneBucket.listKeys(keyPrefix + "a/", null));
+    assertEquals(expectedMessage, exception.getMessage());
+    exception = assertThrows(OMException.class,
+        () -> ozoneBucket.listKeys(keyPrefix + "a///b2///", null));
```
Comment (aswinshakil) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java:299]: Should this be just "a/b2/"?

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java [hunk 2]
```diff
@@ -655,7 +698,6 @@ public void testListStatusOnRoot() throws Exception {
   // based on TestOzoneFileSystem:testListStatusOnLargeDirectory
   public void testListStatusOnLargeDirectory() throws Exception {
     Path root = new Path("/");
-    deleteRootDir(); // cleanup
```
Comment (aswinshakil) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java:658]: Do we not need the cleanup anymore? I believe this is to clean up from the previous test run right?

**Review by aswinshakil — COMMENTED** [review]
Overall LGTM+1. Have some comments below. Thanks for the contribution @hemantk-12

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java (hunk 2 above)
Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java:658]: "this is to clean up from the previous test run right" is my understanding as well.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java (hunk 1 above)
Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java:299]: I kept it as it is here: https://github.com/apache/ozone/blob/master/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java#L280

Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshotFileSystem.java]: Changed it to "a/b2/".

**Review by aswinshakil — APPROVED** [review]

**Comment by hemantk-12** [comment]
Thank for the review @aswinshakil

**Pull request merged**